File: src/util/Locatable.java
package util;

import java.io.File;
import static util.file.UtilKt.childOf;

public interface Locatable {

	File getLocation();

	default File getResource(String path) {
		return childOf(getLocation(), path);
	}

	File getUserLocation();

	default File getUserResource(String path) {
		return childOf(getUserLocation(), path);
	}

}

File: src/sp/it/pl/util/Locatable.kt
package sp.it.pl.util

import sp.it.pl.util.file.div
import java.io.File

interface Locatable {

    val location: File

    val userLocation: File

    @JvmDefault
    fun getResource(path: String): File = location/path

    @JvmDefault
    fun getUserResource(path: String): File = userLocation/path

}